zerver/views/registration.py

```diff
@@ -137,7 +137,7 @@ def accounts_register(request):
             form = RegistrationForm(
                 initial={'full_name': hesiod_name if "@" not in hesiod_name else ""})
             name_validated = True
-        elif settings.POPULATE_PROFILE_VIA_LDAP:
+        elif settings.POPULATE_PROFILE_VIA_LDAP and not realm_creation:
```

Comment:
I guess I'm not totally satisfied with the explanation; is the story that `form.is_valid()` for the registration form passes if this is set and there's no ToS, and that's a problem for realm creation?  

If so, maybe we just want `form.is_valid` to not pass if `realm_creation` is True and the extra fields aren't passed in?  That way, the user would still get their name automatically set by the LDAP setup.